addons/sms/models/mail_thread.py

```diff
@@ -240,7 +240,7 @@ def _notify_thread(self, message, msg_vals=False, notify_by_email=True, **kwargs
         self._notify_record_by_sms(message, recipients_data, msg_vals=msg_vals, **kwargs)
         return recipients_data
 
-    def _notify_record_by_sms(self, message, recipients_data, msg_vals=False,
+    def _notify_record_by_sms(self, message, recipients_data, msg_vals=False, sms_content=None,
```

Comment:
is the `sms_content` really needed ?
(if we have no use case where the fallback is not enough I vote to keep the API simple, I run `test_message_post_sms_from_body` and using only the fallback and the test is green)

Reply:
mhe, so we add an option to manage 1% of the use case ? (even less ?)

Reply:
As we ended up reminding ourselves, we do need to ba able to distinguish sms sent from raw_text from the one after html conversion.
In case of raw text, html tags should be escaped, in case of notif rendered from HTML, HTML should be parsed and the result converted to plaintext.
With special consideration for the links so they are in tests, but it's not only for that.

Reply:
What about trying to be a bit more "generic" and saying
  * body should be the html-friendly version of the content (which means calling plaintext2htmlbutnotreallythesameaspreviouslydefined)
  * body_plaintext optional parameter would be the text version of the body. It was a real parameter a few years ago until we choose to remove it. It could be back from the dead, and it feels more explicit to me: you have the html version, the text version, and hop.

It will be propagated to '_notify_thread' (as not matching any mail.message field), and would be more generic than 'sms_content', while being usable in other context (notably whatsapp, I think we should investigate it someday).

It could be done directly in mail.thread (compute body / body_plaintext if not given, based on the other one). That way sms would call message_post with body_plaintext=body for example. Notification methods would have to use msg_vals['body'] for body, body_plaintext for plaintext version of body.

One issue (that you have here anyway) is that we may call notification methods manually and we always have to fallback on manual computation, which is a bit sad but I don't have much solution.

Reply:
I agree with this, at least the parts I understood (except that the conversion could realistically only go one way).

`sms_content` is the stable-proof version of body_plaintext.

Reply:
Looking at it again, seems ok for me, especially for stable :) it allows to clearly separate content. What do you think @std-odoo  ?